api/src/main/java/org/openmrs/api/context/Credentials.java

```diff
@@ -0,0 +1,29 @@
+/**
+ * This Source Code Form is subject to the terms of the Mozilla Public License,
+ * v. 2.0. If a copy of the MPL was not distributed with this file, You can
+ * obtain one at http://mozilla.org/MPL/2.0/. OpenMRS is also distributed under
+ * the terms of the Healthcare Disclaimer located at http://openmrs.org/license.
+ *
+ * Copyright (C) OpenMRS Inc. OpenMRS is a registered trademark and the OpenMRS
+ * graphic logo is a trademark of OpenMRS Inc.
+ */
+package org.openmrs.api.context;
+
+/**
+ * Authentication schemes define and require their own credentials.
+ * Any client authenticating against a given scheme must supply appropriate credentials.
+ */
+public interface Credentials {
+	
+	/**
+	 * @return The authentication scheme that should be used with those credentials.
+	 */
+	public String getAuthenticationScheme();
+	
+	/**
+	 * A string that identifies the owner of the credentials by a name.
+	 * Typically in the case of a user the client would simply be identified with its username. 
+	 */
+	public String getClientName();
+	
```

Comment:
How do you plan to deal with this? "Credentials will contains scheme-specific properties"

Reply:
@dkayiwa this quote comes from @bmamlin's suggested JavaDoc on the ticket's description. I think the point is that devs/implementers can keep extending `Credentials` implementations (such as extending `UserPassCredentials` for example) or can provide their own implementations of `Credentials`. Either way they will add members and methods that are useful to their custom auth. schemes. We can't anticipate as of yet what will be done, we can just leave the design open enough.